test/Benchmarks/OrleansBenchmarks/MapReduce/MapReduceBenchmark.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.TestingHost;
+using OrleansBenchmarkGrains.MapReduce;
+using OrleansGrainInterfaces.MapReduce;
+
+namespace OrleansBenchmarks.MapReduce
+{
+    public class MapReduceBenchmark
+    {
+        private static TestCluster _host;
+        private readonly int _intermediateStagesCount = 15;
+        private readonly int _pipelineParallelization = 4;
+        private readonly int _repeats = 50000;
+        private int _currentRepeat = 0;
+
+        [Setup]
+        public void BenchmarkSetup()
+        {
+            TestClusterOptions.DefaultTraceToConsole = false;
+            var options = new TestClusterOptions(1);
```

Comment:
Single silo will mean all grain calls are local, bypassing much of the cost.  Not sure this will result in a meaningful benchmark.